src/main/java/org/testng/reporters/FailedReporter.java

```diff
@@ -54,84 +55,50 @@ protected void generateFailureSuite(XmlSuite xmlSuite, ISuite suite, String outp
     failedSuite.setName("Failed suite [" + xmlSuite.getName() + "]");
     m_xmlSuite= failedSuite;
 
-    Map<String, XmlTest> xmlTests= Maps.newHashMap();
-    for(XmlTest xmlT: xmlSuite.getTests()) {
-      xmlTests.put(xmlT.getName(), xmlT);
-    }
-
     Map<String, ISuiteResult> results = suite.getResults();
 
-    synchronized(results) {
```

Comment:
I think we need to keep the synchronized: https://github.com/cbeust/testng/commit/51043eea12cf3f28d291eda5d6c6d4b07b9dedeb